indexing-hadoop/src/main/java/io/druid/indexer/DetermineHashedPartitionsJob.java

```diff
@@ -324,12 +324,9 @@ protected void reduce(
             HyperLogLogCollector.makeCollector(ByteBuffer.wrap(value.getBytes(), 0, value.getLength()))
         );
       }
-      Optional<Interval> intervalOptional = config.getGranularitySpec().bucketInterval(new DateTime(key.get()));
 
-      if (!intervalOptional.isPresent()) {
-        throw new ISE("WTF?! No bucket found for timestamp: %s", key.get());
-      }
-      Interval interval = intervalOptional.get();
+      Interval interval = config.getGranularitySpec().getSegmentGranularity().bucket(new DateTime(key.get()));
```

Comment:
ah right. didn't think of non-contiguous intervals. they are not replaceable.

re: adding this `if (determineIntervals)` statement in the reducer.
The mapper already [throws an exception](https://github.com/praveev/druid/blob/9a88551d6500b52cc984bc3fe9df5f6ae399387a/indexing-hadoop/src/main/java/io/druid/indexer/DetermineHashedPartitionsJob.java#L270) in case it doesn't find an interval for an inputRow timestamp. So i think it is safe to assume that the timestamp will always map to an interval in the reducer? Apparently, that's how it was [originally](https://github.com/druid-io/druid/pull/3674/files#diff-b62980eee0e8bf2eac0a6bf8466b6c74L326)  before it was change in the PR linked to the issue